src/App.tsx

```diff
@@ -27,7 +27,17 @@ import { globalStyles } from './stitches.config';
 import './App.css';
 
 function getLibrary(provider: any): Web3Provider {
-  const library = new Web3Provider(provider);
+  // This checks specifically whether the provider is
+  // an instance of a Web3Provider by checking the existence of this
+  // uniquely named method. Normally, we would want to use `instanceof`
+  // to check if a provider conforms to a specific class, but because
+  // some providers are injected into the window from other contexts,
+  // this check will fail, since those providers aren't derived from
+  // the same prototype tree.
+  const library =
+    typeof provider.jsonRpcFetchFunc !== 'undefined'
```

Comment:
Why check this specific field?

Reply:
@META-DREAMER Our code always expects a `Web3Provider`-wrapped provider for some reason. This is the method that actually communicates onchain, so I figure since we can't go off prototype interitance, we can at least disabiguate a little bit based on the functional piece of the API that is at least a little distinctive, and failing that, named reasonably well such that we can expect an alternative implementation to be functionally equivalent.

Reply:
If there was anything at least a bit more introspective available, I'd gladly use it.